oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elasticsearch/facets/ElasticFacetHelper.java

```diff
@@ -58,32 +58,28 @@ public static ElasticsearchFacets getAggregates(ElasticsearchSearcher searcher,
         return elasticsearchFacets;
     }
 
-    public static List<String> getAccessibleDocIds(SearchHit[] searchHits, Filter filter) throws UnsupportedEncodingException {
+    public static List<String> getAccessibleDocIds(SearchHit[] searchHits, Filter filter) {
         List<String> accessibleDocs = new LinkedList<>();
         for (SearchHit searchHit : searchHits) {
-            String id = searchHit.getId();
-            String path = idToPath(id);
+            final Map<String, Object> sourceMap = searchHit.getSourceAsMap();
+            String path = (String) sourceMap.getOrDefault(FieldNames.PATH, searchHit.getId());
```

Comment:
Technically, we will never use it since `:path` gets always persisted. I have replaced the `getOrDefault` call with a `get` to remove any ambiguity on that.